src/main/java/athena/Athena.java

```diff
@@ -56,8 +56,17 @@ public void runProgram() {
                 allocator.runAllocate();
                 athenaUi.printNewline();
                 athenaUi.printUserInputIndicator();
+
                 inputString = athenaUi.detectInput(input);
-                isExit = logicManager.execute(inputString);
+                TaskList taskList = storage.loadTaskListData();
+                Command userCommand = parser.parse(inputString, taskList);
+                userCommand.execute(taskList, athenaUi);
+
+                TimeAllocator timeAllocator = new TimeAllocator(taskList);
```

Comment:
Maybe we can have a `private TimeAllocator timeAllocator` and put this `timeAllocator = new TimeAllocator(taskList)` in the constructor, since the same `TaskList` object is used everytime? I think it looks nicer than initializing a `TimeAllocator` every time in the loop.